### DISCUSSION TITLE
[Desktop] macOS Arrow key navigation

### DISCUSSION THREAD

**Discussion by Sesa1988** [singerdmx/flutter-quill#172] [Desktop]
My issue is about [Desktop]

Hi, I have following issue on macOS.

**Steps to reproduce**
- Create a Button above the Quill Editor
- Create Quill Editor
- Create QuillBar below Quill Editor
- Press multiple times Down or Up arrow keys

**Result**
Focus is going out of the editor

**Expected Result**
Up and Down arrows keys navigating between the editor content

**Info**
Left and Right arrow keys works as expected, just Up and Down is switching the focus between elements on the screen

**Comment by matuella** [comment]
I think the same applies to Tab (it skips to the next visual element down the tree).

I'm not quite sure if this should be a responsibility of this library, although I am in favor of adding at least a property that allows the handling for these raw key inputs.

Thinking about it, there is a way to do this using a `RawKeyboardListener` + `QuillToolbar`, although this library native support for this kind of use-case is debatable - in my opinion, this would be a useful feature for the desktop experience.

**Comment by GyuriMajercsik** [comment]
As a workaround try to insert the following code where your `MaterialApp` is built: 

```
return MaterialApp(
      shortcuts: {
        ...WidgetsApp.defaultShortcuts,
        if (PlatformUtils.isDesktop)
          LogicalKeySet(LogicalKeyboardKey.space): null,
        if (PlatformUtils.isDesktop)
          LogicalKeySet(LogicalKeyboardKey.arrowUp): null,
        if (PlatformUtils.isDesktop)
          LogicalKeySet(LogicalKeyboardKey.arrowDown): null,
        if (PlatformUtils.isDesktop)
          LogicalKeySet(LogicalKeyboardKey.arrowLeft): null,
        if (PlatformUtils.isDesktop)
          LogicalKeySet(LogicalKeyboardKey.arrowRight): null,
      },
....
```

**Reply by devtud** [reply]
My workaround (tested on Linux only) was setting shortcuts to:

```
      shortcuts: {
        ...WidgetsApp.defaultShortcuts,
        if (Platform.isLinux || Platform.isWindows || Platform.isMacOS)
          LogicalKeySet(LogicalKeyboardKey.space): const ActivateIntent(),
        if (Platform.isLinux || Platform.isWindows || Platform.isMacOS)
          LogicalKeySet(LogicalKeyboardKey.arrowDown): const ActivateIntent(),
        if (Platform.isLinux || Platform.isWindows || Platform.isMacOS)
          LogicalKeySet(LogicalKeyboardKey.arrowUp): const ActivateIntent(),
        if (Platform.isLinux || Platform.isWindows || Platform.isMacOS)
          LogicalKeySet(LogicalKeyboardKey.arrowLeft): const ActivateIntent(),
        if (Platform.isLinux || Platform.isWindows || Platform.isMacOS)
          LogicalKeySet(LogicalKeyboardKey.arrowRight): const ActivateIntent(),
      },
```
